### PR TITLE
more python dependencies for gentoo

### PR DESCRIPTION
**Pull request by Alessandro-Barbieri**

### CODE REVIEW

**Comment by allenh1** [comment]
lgtm

**Comment by Alessandro-Barbieri** [comment]
https://github.com/ros/rosdistro/issues/15082 for reference

File: rosdep/python.yaml [hunk 1]
```diff
@@ -2296,6 +2384,7 @@ python-rosdistro:
 python-rosinstall:
   arch: [python2-rosinstall]
   fedora: [python-rosinstall]
+  gentoo: [dev-util/rosinstall]
```
Comment (allenh1) [inline rosdep/python.yaml]: Please change this to `dev-python/rosinstall`. We'll fix that one if it's not working properly.

File: rosdep/python.yaml [hunk 2]
```diff
@@ -2311,6 +2400,7 @@ python-rosinstall:
 python-rosinstall-generator:
   arch: [python2-rosinstall-generator]
   fedora: [python-rosinstall_generator]
+  gentoo: [dev-util/rosinstall_generator]
```
Comment (allenh1) [inline rosdep/python.yaml]: Please change to `dev-python/rosinstall_generator`

File: rosdep/python.yaml [hunk 3]
```diff
@@ -3105,17 +3227,20 @@ python-websocket:
...
   fedora: [python-werkzeug]
+  gentoo: [dev-python/werkzeug]
   ubuntu: [python-werkzeug]
 python-ws4py-pip:
   ubuntu:
     pip:
       packages: [ws4py]
 python-wstool:
   fedora: [python-wstool]
+  gentoo: [dev-util/wstool]
```
Comment (allenh1) [inline rosdep/python.yaml]: please use `dev-python/wstool`

File: rosdep/python.yaml (hunk 2 above)
Comment (Alessandro-Barbieri) [inline rosdep/python.yaml]: Does have something special?

File: rosdep/python.yaml (hunk 3 above)
Comment (Alessandro-Barbieri) [inline rosdep/python.yaml]: It's different from gentoo version?

File: rosdep/python.yaml (hunk 2 above)
Comment (allenh1) [inline rosdep/python.yaml]: I mean, I for one am very strongly opposed to mixing them. We maintain ours, and we don't want the downstream people to be ahead/behind the official repo. If we are behind, please file an issue requesting that the ebuild be updated.

File: rosdep/python.yaml (hunk 3 above)
Comment (allenh1) [inline rosdep/python.yaml]: Maybe not at the moment, but it's not _our_ version.

File: rosdep/python.yaml (hunk 2 above)
Comment (Alessandro-Barbieri) [inline rosdep/python.yaml]: `rosinstall_generator::ros-overlay` it's outdated

**Comment by Alessandro-Barbieri** [comment]
Since these tool are already in gentoo they are to be the default choice

File: rosdep/python.yaml (hunk 2 above)
Comment (allenh1) [inline rosdep/python.yaml]: Then I'll update it

File: rosdep/python.yaml (hunk 1 above)
Comment (Alessandro-Barbieri) [inline rosdep/python.yaml]: Overlay `rosinstall` isn't different from gentoo one

File: rosdep/python.yaml (hunk 2 above)
Comment (Alessandro-Barbieri) [inline rosdep/python.yaml]: And it's isn't different from gentoo one

**Comment by Alessandro-Barbieri** [comment]
Checked  the overlay ebuilds, they don't have anything special apart bad being outdated, dependencies and python issues

**Comment by allenh1** [comment]
@Alessandro-Barbieri while that may be true, my point is that it may cease to be true in the future.

For instance, maybe the Gentoo people decide that, since the official install method has a much larger set of packages and is updated far more frequently, that they will be more lax about maintaining their ebuilds.

I feel it would be far more simple to make the official set of installs self-contained, so it is my opinion that we should refrain from using the downstream version wherever possible.

**Comment by Alessandro-Barbieri** [comment]
Gentoo is about choice so change the category name and let the user choose if using the overlay or not (isn't needed for now since downstream have the tools for a source based install)

**Comment by Alessandro-Barbieri** [comment]
'Now' they have the same tools as here, if in future some ebuilds will need change, you can change it (and if you retain the same category the 'upgrade' will be seamless by just adding the overlay and pull the fixed version)

**Comment by allenh1** [comment]
> 'Now' they have the same tools as here, if in future some ebuilds will need change, you can change it (and if you retain the same category the 'upgrade' will be seamless by just adding the overlay and pull the fixed version)

In hindsight, that would be the solution. But, the same point could be made for those using the `dev-python` versions, which is the method mentioned in the official install guide.

This isn't about a _mine_ or _theirs_ thing, this is a consistency thing. The goal here is to keep everything consistent and without mixing our repo and their repo, regardless of content.

The last reason I have for not changing the name is that our ebuilds actually predate the official ones. I am also not sure why we should go to the `dev-util` category, since these are python api's which I think should belong in the `dev-python` category.

I'm not trying to call you wrong here, as you are by no means wrong. I am simply clarifying that it would be my preference to use the existing category, rather than change to another one.

**Comment by Alessandro-Barbieri** [comment]
The guide can be changed easily, old users can transition without much trouble.

**Comment by Alessandro-Barbieri** [comment]
IMHO they weren't put in the `dev-python` category because these tools aren't python libraries
for example `gentoolkit` is written in python but it's in `app-portage`

**Comment by Alessandro-Barbieri** [comment]
@allenh1 ok? I removed the category so it will work with or without the overlay

**Comment by allenh1** [comment]
So, that doesn't help me with ebuild generation... Portage became very unhappy when dependencies didn't have a category attached to them... Otherwise that would be the best solution.

**Comment by mikaelarguedas** [comment]
As a non Gentoo expert and not familiar with the exact meaning or use of categories, I'd vouch for having a behavior as close as the other systems. For Ubuntu or Debian, ROS packages, even if available upstream are downloaded from our apt repo with our latest released version. So whichever category reproduces this behavior is the one that should be used IMO.
If the Gentoo community wants to support installing ROS using their version of these packages (that's awesome!) but doesn't have to be reflected here.

**Comment by Alessandro-Barbieri** [comment]
There is not 'our' or 'their' version: only upstream version matter and gentoo provides exactly what can be found upstream. Gentoo *does not* modify code of what is found upstream, so why worry? Why provide the same thing in an incompatible way? The ebuilds in the overlay doesn't add *nothing* apart of being written worst than downstream and providing outdated version.

**Comment by Alessandro-Barbieri** [comment]
@allenh1 how to you query portage to generate the ebuilds?

**Comment by allenh1** [comment]
I wrote a custom system (note: not completed): https://github.com/allenh1/ebuild-generator

**Comment by Alessandro-Barbieri** [comment]
Crazy idea here: for example create a fake `dev-util/rosinstall` that rdepends on `dev-python/rosinstall`

**Comment by allenh1** [comment]
While that would work, that's still not my preference.

**Comment by Alessandro-Barbieri** [comment]
@allenh1 

>Otherwise that would be the best solution.

Trys this code (you'll need `eix`)

```python
#check if pkg does not contain the category an then searches in the overlay for the <category>/<name> matching pkg

 if "/" in pkg:
    pkg=subprocess.Popen("eix --exact %s --in-overlay ros-overlay --only-names" %pkg, shell=True,  stdout=subprocess.PIPE, stderr=subprocess.STDOUT).stdout.read().decode('
    ...: ascii')
```

**Comment by allenh1** [comment]
I think we're going to update the existing ones, keeping the `dev-python` prefix. The primary reason here is that I don't want to break existing installations/confuse portage when ebuilds suddenly disappear.

If we change later, that's fine, but I'd rather get the dependencies pulled in and later have a separate discussion about the `dev-python`/`dev-util` issue. @Alessandro-Barbieri is that fair?

**Comment by Alessandro-Barbieri** [comment]
>I don't want to break existing installations/confuse portage when ebuilds suddenly disappear.

You will not break anything, if you use that snippet it will (when you generate the ebuilds) convert, for example, `rosinstall` in `dev-python/rosinstall`
`eix` is super fast (really: `0,211 s` for that command) and will run only for the few packages listed without a category

**Comment by Alessandro-Barbieri** [comment]
Well, do what you want, I want to try to installing using the old method.
Please don't nuke this pull request, it's possible to add a pull request after?

**Comment by allenh1** [comment]
Well, do what you want, I want to try to installing using the old method.
Please don't nuke this pull request, it's possible to add a pull request after?

That would be greatly preferred by me

**Comment by Alessandro-Barbieri** [comment]
@allenh1 since workaround exist I made the changes.
Ready to be merged.

File: rosdep/python.yaml (hunk 1 above)
Comment (Alessandro-Barbieri) [inline rosdep/python.yaml]: done

File: rosdep/python.yaml (hunk 2 above)
Comment (Alessandro-Barbieri) [inline rosdep/python.yaml]: done

File: rosdep/python.yaml (hunk 3 above)
Comment (Alessandro-Barbieri) [inline rosdep/python.yaml]: done

**Review by Alessandro-Barbieri — COMMENTED** [review]
Time to merge

**Comment by Alessandro-Barbieri** [comment]
@allenh1 can you rewiev the PR? Does it need other changes?

**Comment by Alessandro-Barbieri** [comment]
@allenh1 @mikaelarguedas why is taking so long? It's something missing?

**Review by mikaelarguedas — APPROVED** [review]

**Comment by allenh1** [comment]
+1

**Pull request merged**